### PR TITLE
hlcy, 1ref, mnvv, cpofp, solar, ssrun, vig, vbdsf, vddsf

### PR DESCRIPTION
**Pull request by cshartsough**
new graphics for:

- min hlcy (4 levels)
- storm relative hlcy (2 levels),
- 1 hr reflectivity
- frozen precip %
- incoming SW/Direct/Diffuse radiation
- mean vertical velocity
- storm surface runoff
- vertically-integrated graupel.

pylint and pytest passed.  plot examples to follow.

### CODE REVIEW

**Comment by cshartsough** [comment]
plot examples. 14 plots but only 3 new color tables.  ;-)

curious about mean vertical velocity.  Wondering if small up and down values are close together so they cover each other up and the color that shows depends on the order of plotting.  can discuss it here or in a meet-up.

[web graphics - PR41.pptx](https://github.com/NOAA-GSD/adb_graphics/files/4979416/web.graphics.-.PR41.pptx)

File: adb_graphics/default_specs.yml [hunk 1]
```diff
@@ -55,6 +55,31 @@
 #      anchor needs to be applied before another section.
 #
 
+1hsm: # Helicity
```
Comment (christinaholtNOAA) [inline adb_graphics/default_specs.yml:58]: How does this helicity differ from the hlcy section? There is already a sr01 level in that section.

File: adb_graphics/default_specs.yml [hunk 2]
```diff
@@ -55,6 +55,31 @@
...
+  sfc: &hlcy # 0-1 km Storm Relative Helicity
+    clevs: [25, 50, 100, 150, 200, 250, 300, 400, 500, 600, 700, 800]
+    cmap: gist_ncar
+    colors: hlcy_colors
+    layer: 0
+    ncl_name: HLCY_P0_2L103_GLC0
+    ticks: 0
+    title: 0-1 km Storm Relative Helicity
+    unit: m2/s2
+1ref: # Composite reflectivity
```
Comment (christinaholtNOAA) [inline adb_graphics/default_specs.yml:68]: Don't we have another Composity Reflectivity section? If yes, how does this differ?

File: adb_graphics/default_specs.yml [hunk 3]
```diff
@@ -55,6 +55,31 @@
...
+  1000m: &refl
+    clevs: !!python/object/apply:numpy.arange [5, 76, 5]
+    cmap: NWSReflectivity
+    colors: cref_colors
+    ncl_name: REFD_P0_L103_GLC0
+    ticks: 5
+    title: 1 km agl Reflectivity
+    unit: dBZ
+3hsm: # Helicity
```
Comment (christinaholtNOAA) [inline adb_graphics/default_specs.yml:77]: Same comment as 1hsm above.

File: adb_graphics/default_specs.yml [hunk 4]
```diff
@@ -163,13 +188,19 @@ cloudcover:
...
-    clevs: !!python/object/apply:numpy.arange [5, 76, 5]
-    cmap: NWSReflectivity
-    colors: cref_colors
+  sfc:
+    <<: *refl
     ncl_name: REFC_P0_L10_GLC0
-    ticks: 5
-    unit: dB
+    title: Composite Reflectivity
+cpofp:
```
Comment (christinaholtNOAA) [inline adb_graphics/default_specs.yml:195]: What do these letters mean?

File: adb_graphics/default_specs.yml [hunk 5]
```diff
@@ -394,6 +452,15 @@ ltg3: # Lightning Threat (LTG1 + LTG2)
     ticks: 0
     title: Lightning Threat (comb of LTG1 and LTG2)
     unit: flashes / km^2 / 5 min
+mnvv: # Mean Vertical Veolcity
```
Comment (christinaholtNOAA) [inline adb_graphics/default_specs.yml:455]: Can this live as a level under vvel below?

File: adb_graphics/default_specs.yml [hunk 6]
```diff
@@ -715,11 +796,30 @@ v:
   925mb: *ua_vwind
   ua:
     <<: *ua_vwind
+vbdsf: # Incoming Direct Radiation
```
Comment (christinaholtNOAA) [inline adb_graphics/default_specs.yml:799]: Better name here and vddsf?

File: adb_graphics/default_specs.yml (hunk 1 above)
Comment (cshartsough) [reply adb_graphics/default_specs.yml:58]: > Wow! Another LARGE set down!
> 
> I noticed that all the helicity plots needed the wind barbs turned on.

Some of them wind shear and others plot storm motion, all as wind barbs.  It should be easy to turn that on if we can distinguish which winds to plot.

Comment (cshartsough) [reply adb_graphics/default_specs.yml:58]: These are the same, I was going by the spreadsheet name and didn't notice the placeholders under hlcy.  I can move them if we prefer to have them under hlcy.

File: adb_graphics/default_specs.yml (hunk 2 above)
Comment (cshartsough) [reply adb_graphics/default_specs.yml:68]: my mistake, forgot to change the label after cut-and-paste.  1ref is the reflectivity 1 km AGL.

File: adb_graphics/default_specs.yml (hunk 4 above)
Comment (cshartsough) [reply adb_graphics/default_specs.yml:195]: Categorical Percentage Of Frozen Precipitation.  Seems to be standard at least for NCEP and GSL.

I moved the name comment up a line from "sfc:" line below it.

File: adb_graphics/datahandler/grib.py [hunk 7]
```diff
@@ -328,26 +328,35 @@ def values(self, level=None, name=None, **kwargs) -> np.ndarray:
...
-                lev_val, lev_unit = self.numeric_level()
-                lev_val = lev_val * 100. if lev_unit == 'mb' else lev_val
-
-                # The index of the reqested level
+            dim_name = spec.get('vertical_level_name',
+                                field.dimensions[0])
+            print('grib.py: dim_name = ', dim_name)
+            levs = self.contents.variables[dim_name][::]
+            print('grib.py: levs = ', levs)
+
```
Comment (christinaholtNOAA) [inline adb_graphics/datahandler/grib.py:336]: Are the print statements meant for debugging, or do we want to leave them here? I see some below, too.

File: adb_graphics/specs.py [hunk 8]
```diff
@@ -352,7 +396,7 @@ def vort_colors(self) -> np.ndarray:
 
         grays = cm.get_cmap('Greys', 2)([0])
         ncar = cm.get_cmap(self.vspec.get('cmap'), 128) \
-                          ([15, 18, 20, 25, 50, 60, 70, 80, 85, 90, 100, 120])
+                          ([15, 18, 20, 25, 50, 60, 70, 80, 83, 90, 100, 120])
```
Comment (christinaholtNOAA) [inline adb_graphics/specs.py:399]: I am noticing that vort_colors, vis_colors, and rh_colors are all the same code. Can we name this function something generic and just have it here once? If there are other occurrences of the same one, let's also combine those in, too.

**Pull request merged**